src/main/java/io/github/classgraph/ClasspathElementZip.java

```diff
@@ -281,7 +281,7 @@ void open(final WorkQueue<ClasspathEntryWorkUnit> workQueue, final LogNode log)
      * @return the resource
      */
     private Resource newResource(final FastZipEntry zipEntry, final String pathRelativeToPackageRoot) {
-        return new Resource(this, zipEntry.uncompressedSize) {
+        return new Resource(this, zipEntry.uncompressedSize, zipEntry.lastModified, zipEntry.posixFilePermissions) {
```

Comment:
`ClasspathElementDir` needs corresponding changes, and then I think the PR is ready to be merged. I can jump in and do that, or let me know if you're on a roll :-)

Reply:
Cool. I’ll get that done tomorrow.

Reply:
In that case, I'll go ahead and make the change in `ClasspathElementDir`, that way I can push out a release tonight, and you'll have a jar you can use in Maven Central by tomorrow.